packages/lib/src/vis/shared/VisCanvas.tsx

```diff
@@ -68,26 +68,24 @@ function VisCanvas(props: PropsWithChildren<Props>) {
         resize={{ debounce: { scroll: 20, resize: 200 }, scroll: false }} // https://github.com/pmndrs/react-three-fiber/discussions/1906
       >
         <ambientLight />
-        <AxisSystemProvider
+        <VisCanvasProvider
           visRatio={visRatio}
           abscissaConfig={abscissaConfig}
           ordinateConfig={ordinateConfig}
           floatingToolbar={floatingToolbar}
         >
-          <InteractionsProvider>
-            <AxisSystem
-              axisOffsets={axisOffsets}
-              title={title}
-              showAxes={showAxes}
-            />
-            {children}
-            <ViewportCenterer />
-            <RatioEnforcer visRatio={visRatio} />
-            {raycasterThreshold !== undefined && (
-              <ThresholdAdjuster value={raycasterThreshold} />
-            )}
-          </InteractionsProvider>
-        </AxisSystemProvider>
+          <AxisSystem
+            axisOffsets={axisOffsets}
+            title={title}
+            showAxes={showAxes}
+          />
+          <InteractionsProvider>{children}</InteractionsProvider>
```

Comment:
Moving `InteractionsProvider` around `children`, since the other components don't need it.